### PR TITLE
[MTSRE-528] serviceMonitor for AddonCR.

### PR DESCRIPTION
**Pull request by sugarraysam**
# py-mtcli

## Description

Feature parity with the addon-operator.

Depends on https://github.com/openshift/addon-operator/pull/186

### CODE REVIEW

File: tests/sss/test_federated_metrics.py [hunk 1]
```diff
@@ -69,6 +69,8 @@ def test_namespace_and_servicemonitor_v2(data):
     env = data.draw(custom_strategies.environment())
     addon = data.draw(custom_strategies.addon(env))
     addon.metadata["monitoring"] = {
+        "portName": data.draw(custom_strategies.k8s_name()),
+        "namespace": data.draw(custom_strategies.namespace()),
```
Comment (sandhyadalavi) [inline tests/sss/test_federated_metrics.py:73]: can we add matchlabels and matchnames here?

**Review by apahim — CHANGES_REQUESTED** [review]
Can we remove the AddonCR from the SelectorSyncSet instead? It's not used anywhere, it's just causing confusion, and the feature parity work should happen here:

https://github.com/openshift/addon-operator/blob/main/internal/controllers/addon/phase_ensure_monitoring_federation.go

**Comment by apahim** [comment]
I see you hve it here: https://github.com/openshift/addon-operator/pull/186/files

So, yeah, it all good, I believe. Unless OCM needs to pass some parameter through, as it's OCM who creates the Addon CR, not this SSS.

**Comment by sugarraysam** [comment]
@apahim let me close this PR then and create a new one for the removal of the Addon CR.

**Pull request closed without merging**